### PR TITLE
feat: Items/BOQ in Task

### PR DESCRIPTION
**Pull request by noahjacob**
### Features
<details>

<summary>Task Items Table</summary>

image

</details>

- User can now add task items.

- Items can be fetched from a BOM (Similar to what is in **Stock Entry).**

image

- Material Issue/Transfers can be created via Task.

image

<details>

<summary>Material Transfer</summary>  
<p></p>

- The **Project Warehouse** set in the Task is set as the target warehouse in the **Material Transfer**.  
- The Transferred qty is updated in the **Task Item Table**.

boq_trans

</details>

<details>

<summary>Material Issue</summary>
<p></p>

- Only remaining required qty are mapped onto the Stock Entry.
- The Issued qty is updated in the **Task Item Table**.

boq_issue

</details>

- Total Estimated and Actual Costs are also calculated.

image

- Added a link to **Stock Entry** in the dashboard.

image

Docs: https://docs.erpnext.com/docs/v13/user/manual/en/projects/tasks/edit?wiki_page_patch=8c37e4a782

### CODE REVIEW

**Comment by mujeerhashmi** [comment]
We can also have BOM Operation for Tasks. So that we will have the provision to track equipment costs.

**Comment by muchai** [comment]
Is this not tied to projects/project tasks?
It's a very exciting feature, and I thought it's applicable in construction industry, steel structures firms...

File: erpnext/projects/doctype/task/task.js [hunk 1]
```diff
@@ -57,5 +105,87 @@ frappe.ui.form.on("Task", {
...
+				}
+			},
+			callback: function (r) {
+				frappe.model.set_value(cdt, cdn, 'basic_rate', r.message);
+				row.estimated_cost = flt(row.qty) * flt(row.basic_rate);
+				frm.events.calculate_total_estimated_costs(frm);
+			},
+		});
+	},
+	calculate_total_estimated_costs(frm) {
```
Comment (rmehta) [inline erpnext/projects/doctype/task/task.js:141]: all business logic should be serverside

**Review by rmehta — CHANGES_REQUESTED** [review]

**Comment by deepeshgarg007** [comment]
Closing this due to no action

**Pull request closed without merging**